django/cantusdb_project/main_app/templates/browse_chants.html

```diff
@@ -52,79 +52,84 @@ <h3>Browse Chants</h3>
-                        <tr>
-                            <td class="text-wrap" style="text-align:center" title="{{ chant.source.title }}">{{ chant.source.siglum|default:"" }}</td>
-                            <td class="text-wrap" style="text-align:center"><b>{{ chant.folio|default:"" }}</b></td>
-                            <td class="text-wrap" style="text-align:center">{{ chant.c_sequence|default_if_none:"" }}</td> {# default_if_none: sometimes, c_sequence is 0, and should still be displayed #}
-                            <td class="text-wrap">
-                                <a href="{% url 'chant-detail' chant.id %}"><b>{{ chant.incipit|default:"" }}</b></a>
-                                <p>{{ chant.manuscript_full_text_std_spelling|default:"" }}<br>
-                                {% if chant.volpiano %}
-                                    <span style="font-family: volpiano; font-size:25px">{{ chant.volpiano|default:"" }}</span>
-                                {% endif %}
-                                </p>
-                            </td>
-                            <td class="text-wrap" style="text-align:center">
-                                {% if chant.feast %}
-                                    <a href="{% url 'feast-detail' chant.feast.id %}" title="{{ chant.feast.description }}">{{ chant.feast.name|default:"" }}</a>
-                                {% endif %}
-                            </td>
-                            <td class="text-wrap" style="text-align:center">
-                                {% if chant.office %}
-                                    <a href="{% url 'office-detail' chant.office.id %}" title="{{ chant.office.description }}">{{ chant.office.name|default:"" }}</a>
-                                {% endif %}
-                            </td>
-                            <td class="text-wrap" style="text-align:center">
-                                {% if chant.genre %}
-                                    <a href="{% url 'genre-detail' chant.genre.id %}" title="{{ chant.genre.description }}">{{ chant.genre.name|default:"" }}</a>
-                                {% endif %}
-                            </td>
-                            <td class="text-wrap" style="text-align:center">{{ chant.position|default:"" }}</td>
-                            <td class="text-wrap" style="text-align:center">
-                                <a href="{{ chant.get_ci_url }}" target="_blank">{{ chant.cantus_id|default:"" }}</a>
-                            </td>
-                            <td class="text-wrap" style="text-align:center">{{ chant.mode|default:"" }}</td>
-                            <td class="text-wrap" style="text-align:center">
-                                {% if chant.image_link %}
-                                    <a href="{{ chant.image_link }}" target="_blank">Image</a>
+            {% with exists_on_cantus_ultimus=source.exists_on_cantus_ultimus%}
```

Comment:
> do you need to check this? Won't it be the same for every chant in the source...so you could just check from the source?

You're right, and I actually put a source-level check earlier in the code here at line 55. I guess i just forgot I put it there.